test/envtest/konnect_entities_suite_test.go

```diff
@@ -111,3 +113,77 @@ func testNewKonnectEntityReconciler[
 		}
 	})
 }
+
+const (
+	// testCpID is a test Konnect Control Plane ID.
+	testCpID = "cp-id"
+)
+
+// prepareValidKonnectCP creates a KonnectGatewayControlPlane with a reference to a KonnectAPIAuthConfiguration with
+// statuses that make them both usable in tests that require a reference to a valid Konnect Control Plane.
+func prepareValidKonnectCP(ctx context.Context, t *testing.T, ns string, cl client.Client) (cpName string) {
+	auth := &konnectv1alpha1.KonnectAPIAuthConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      uuid.NewString(),
+			Namespace: ns,
+		},
+		Spec: konnectv1alpha1.KonnectAPIAuthConfigurationSpec{
+			Type:      konnectv1alpha1.KonnectAPIAuthTypeToken,
+			Token:     "kpat_test",
+			ServerURL: "127.0.0.1",
+		},
+	}
+	require.NoError(t, cl.Create(ctx, auth))
+
+	// We cannot create KonnectAPIAuthConfiguration with specified status, so we update the status after creating it.
+	auth.Status = konnectv1alpha1.KonnectAPIAuthConfigurationStatus{
+		OrganizationID: "org-1",
+		ServerURL:      "127.0.0.1",
+		Conditions: []metav1.Condition{
+			{
+				Type:               conditions.KonnectEntityAPIAuthConfigurationValidConditionType,
+				ObservedGeneration: auth.GetGeneration(),
+				Status:             metav1.ConditionTrue,
+				Reason:             conditions.KonnectEntityAPIAuthConfigurationReasonValid,
+				LastTransitionTime: metav1.Now(),
+			},
+		},
+	}
+	require.NoError(t, cl.Status().Update(ctx, auth))
+
+	// Create KonnectGatewayControlPlane.
+	cpName = uuid.NewString()
```

Comment:
Nit: I suggest we use `GenerateName` with a meaningful prefix here rather than long uuids which are cumbersome to work with when debugging tests. WDYT?